### ISSUE TITLE
Invite, Reject or Mail forward does not work

### ISSUE THREAD

**Issue by cbleek** [cross-solution/YAWIK#619]
You'll get a  "getTimestamp() on null" error.

![Auswahl_999(845)](https://user-images.githubusercontent.com/858642/103550649-843d9500-4ea9-11eb-8d39-f80c068fa05d.png)

```
[2021-01-04 16-20-58] Error: Call to a member function getTimestamp() on null in /var/www/staging/releases/54/module/Applications/src/Listener/StatusChange.php:108  @  https://stag-01-04--16-19--49ac2c0af4.html
[2021-01-04 16-22-29] Error: Call to a member function getTimestamp() on null in /var/www/staging/releases/54/module/Applications/src/Mail/Forward.php:66  @  https://staging.yawik.-solution.de  @@  error--2021-01-04--16-22--32980c7c01.html
```

@kilip , @TiSiE 

who can watch this?

----

Call to a member function getTimestamp() on null

```
Source file

File: .../module/Applications/src/Mail/Forward.php:66

56:            }
57:            $this->setEncoding('UTF-8');
58:            $subject = /* @translate */ 'Fwd: Application to "%s" dated %s';
59:            if ($this->isTranslatorEnabled()) {
60:                $subject = $this->getTranslator()->translate($subject);
61:            }
62:            $this->setSubject(
63:                sprintf(
64:                    $subject,
65:                    $this->application->getJob()->getTitle(),
66:                    strftime('%x', $this->application->getDateCreated()->getTimestamp())
67:                )
68:            );
69:            $this->generateBody();
70:        }

Call stack

 1. .../54/module/Core/src/Mail/MailService.php:126 source  Applications\Mail\
    Forward->init()

    116:            //   function ($context,$instance) {
    117:            //        if (method_exists($instance, 'setServiceLocator')) {
    118:            //            //$instance->setServiceLocator($this);
    119:            //        }
    120:            //   }
    121:            //);
    122:
    123:            $this->addInitializer(
    124:                function ($context, $instance) {
    125:                    if (method_exists($instance, 'init')) {
    126:                        $instance->init();
    127:                    }
    128:                }
    129:            );
    130:        }
    
 2. .../src/ServiceManager.php:780 source  Core\Mail\MailService->Core\Mail\
    {closure}(arguments)

    770:                throw $exception;
    771:            } catch (Exception $exception) {
    772:                throw new ServiceNotCreatedException(sprintf(
    773:                    'Service with name "%s" could not be created. Reason: %s',
    774:                    $resolvedName,
    775:                    $exception->getMessage()
    776:                ), (int) $exception->getCode(), $exception);
    777:            }
    778:
    779:            foreach ($this->initializers as $initializer) {
    780:                $initializer($this->creationContext, $object);
    781:            }
    782:
    783:            return $object;
    784:        }
    
    #0
    #1

 3. .../src/ServiceManager.php:228 source  Laminas\ServiceManager\
    ServiceManager->doCreate(arguments)

    218:            return $object;
    219:        }
    220:
    221:        /**
    222:         * {@inheritDoc}
    223:         */
    224:        public function build($name, array $options = null)
    225:        {
    226:            // We never cache when using "build"
    227:            $name = isset($this->resolvedAliases[$name]) ? $this->resolvedAliases[$name] : $name;
    228:            return $this->doCreate($name, $options);
    229:        }
    230:
    231:        /**
    232:         * {@inheritDoc}
    
    $resolvedName "Applications/Forward" (20)

      $options

 4. .../src/AbstractPluginManager.php:153 source  Laminas\ServiceManager\
    ServiceManager->build(arguments)

    143:                    throw new Exception\ServiceNotFoundException(sprintf(
    144:                        'A plugin by the name "%s" was not found in the plugin manager %s',
    145:                        $name,
    146:                        get_class($this)
    147:                    ));
    148:                }
    149:
    150:                $this->setFactory($name, Factory\InvokableFactory::class);
    151:            }
    152:
    153:            $instance = empty($options) ? parent::get($name) : $this->build($name, $options);
    154:            $this->validate($instance);
    155:            return $instance;
    156:        }
    157:
    
     $name   "Applications/Forward" (20)

    $options

 5. .../Core/src/Controller/Plugin/Mailer.php:62 source  Laminas\ServiceManager
    \AbstractPluginManager->get(arguments)

    52:                sprintf(
    53:                    'Neither proxied class "%s" nor this class knows of a method called "%s"',
    54:                    get_class($mailService),
    55:                    $method
    56:                )
    57:            );
    58:        }
    59:
    60:        public function get($mailPluginName, array $options = array())
    61:        {
    62:            return $this->getMailService()->get($mailPluginName, $options);
    63:        }
    64:
    65:        public function send(Message $mail)
    66:        {
    
     $name   "Applications/Forward" (20)

    $options

 6. .../Core/src/Controller/Plugin/Mailer.php:84 source  Core\Controller\Plugin
    \Mailer->get(arguments)

    74:            }
    75:            if ($mail instanceof Message) {
    76:                return $this->send($mail);
    77:            }
    78:
    79:            if (is_bool($options)) {
    80:                $sendMail = $options;
    81:                $options  = array();
    82:            }
    83:
    84:            $mail = $this->get($mail, $options);
    85:
    86:            return $sendMail ? $this->send($mail) : $mail;
    87:        }
    88:
    
    $mailPluginName "Applications/Forward" (20)

       $options

 7. inner-code Core\Controller\Plugin\Mailer->__invoke(arguments)

      $mail   "Applications/Forward" (20)

    $options
    $sendMail true

 8. .../src/Controller/AbstractController.php:275 source  call_user_func_array(
    arguments)

    265:         * Otherwise, return the plugin instance.
    266:         *
    267:         * @param  string $method
    268:         * @param  array  $params
    269:         * @return mixed
    270:         */
    271:        public function __call($method, $params)
    272:        {
    273:            $plugin = $this->plugin($method);
    274:            if (is_callable($plugin)) {
    275:                return call_user_func_array($plugin, $params);
    276:            }
    277:
    278:            return $plugin;
    279:        }
    
    $function_name
      $parameters

 9. .../src/Controller/ManageController.php:438 source  Laminas\Mvc\Controller\
    AbstractController->__call(arguments)

    428:            );
    429:
    430:            try {
    431:                $userName    = $this->auth('info')->getDisplayName();
    432:                $fromAddress = $application->getJob()->getContactEmail();
    433:                $mailOptions = array(
    434:                    'application' => $application,
    435:                    'to'          => $emailAddress,
    436:                    'from'        => array($fromAddress => $userName)
    437:                );
    438:                $this->mailer('Applications/Forward', $mailOptions, true);
    439:                $this->notification()->success($params['text']);
    440:            } catch (\Exception $ex) {
    441:                $params = array(
    442:                    'ok' => false,
    
    $method "mailer" (6)

    $params

10. .../Controller/AbstractActionController.php:77 source  Applications\
    Controller\ManageController->forwardAction()

    67:                throw new Exception\DomainException('Missing route matches; unsure how to retrieve action');
    68:            }
    69:
    70:            $action = $routeMatch->getParam('action', 'not-found');
    71:            $method = static::getMethodFromAction($action);
    72:
    73:            if (! method_exists($this, $method)) {
    74:                $method = 'notFoundAction';
    75:            }
    76:
    77:            $actionResponse = $this->$method();
    78:
    79:            $e->setResult($actionResponse);
    80:
    81:            return $actionResponse;
    
11. .../src/Controller/ManageController.php:108 source  Laminas\Mvc\Controller\
    AbstractActionController->onDispatch(arguments)

     98:         */
     99:        public function onDispatch(\Laminas\Mvc\MvcEvent $e)
    100:        {
    101:            $routeMatch = $e->getRouteMatch();
    102:            $action     = $this->params()->fromQuery('action');
    103:
    104:            if ($routeMatch && $action) {
    105:                $routeMatch->setParam('action', $action);
    106:            }
    107:
    108:            return parent::onDispatch($e);
    109:        }
    110:
    111:        /**
    112:         * List applications
    
    $e

12. .../src/EventManager.php:331 source  Applications\Controller\
    ManageController->onDispatch(arguments)

    321:            krsort($listOfListenersByPriority);
    322:
    323:            // Initial value of stop propagation flag should be false
    324:            $event->stopPropagation(false);
    325:
    326:            // Execute listeners
    327:            $responses = new ResponseCollection();
    328:            foreach ($listOfListenersByPriority as $listOfListeners) {
    329:                foreach ($listOfListeners as $listeners) {
    330:                    foreach ($listeners as $listener) {
    331:                        $response = $listener($event);
    332:                        $responses->push($response);
    333:
    334:                        // If the event was asked to stop propagating, do so
    335:                        if ($event->propagationIsStopped()) {
    
    $e

13. .../src/EventManager.php:188 source  Laminas\EventManager\EventManager->
    triggerListeners(arguments)

    178:        public function triggerEvent(EventInterface $event)
    179:        {
    180:            return $this->triggerListeners($event);
    181:        }
    182:
    183:        /**
    184:         * @inheritDoc
    185:         */
    186:        public function triggerEventUntil(callable $callback, EventInterface $event)
    187:        {
    188:            return $this->triggerListeners($event, $callback);
    189:        }
    190:
    191:        /**
    192:         * @inheritDoc
    
      $event
    $callback

14. .../src/Controller/AbstractController.php:103 source  Laminas\EventManager\
    EventManager->triggerEventUntil(arguments)

     93:                $response = new HttpResponse();
     94:            }
     95:            $this->response = $response;
     96:
     97:            $e = $this->getEvent();
     98:            $e->setName(MvcEvent::EVENT_DISPATCH);
     99:            $e->setRequest($request);
    100:            $e->setResponse($response);
    101:            $e->setTarget($this);
    102:
    103:            $result = $this->getEventManager()->triggerEventUntil(function ($test) {
    104:                return ($test instanceof Response);
    105:            }, $e);
    106:
    107:            if ($result->stopped()) {
    
    $callback
      $event

15. .../laminas-mvc/src/DispatchListener.php:139 source  Laminas\Mvc\Controller
    \AbstractController->dispatch(arguments)

    129:
    130:            if ($controller instanceof InjectApplicationEventInterface) {
    131:                $controller->setEvent($e);
    132:            }
    133:
    134:            $request  = $e->getRequest();
    135:            $response = $application->getResponse();
    136:            $caughtException = null;
    137:
    138:            try {
    139:                $return = $controller->dispatch($request, $response);
    140:            } catch (\Throwable $ex) {
    141:                $caughtException = $ex;
    142:            } catch (\Exception $ex) {  // @TODO clean up once PHP 7 requirement is enforced
    143:                $caughtException = $ex;
    
     $request
    $response

16. .../src/EventManager.php:331 source  Laminas\Mvc\DispatchListener->
    onDispatch(arguments)

    321:            krsort($listOfListenersByPriority);
    322:
    323:            // Initial value of stop propagation flag should be false
    324:            $event->stopPropagation(false);
    325:
    326:            // Execute listeners
    327:            $responses = new ResponseCollection();
    328:            foreach ($listOfListenersByPriority as $listOfListeners) {
    329:                foreach ($listOfListeners as $listeners) {
    330:                    foreach ($listeners as $listener) {
    331:                        $response = $listener($event);
    332:                        $responses->push($response);
    333:
    334:                        // If the event was asked to stop propagating, do so
    335:                        if ($event->propagationIsStopped()) {
    
    $e

17. .../src/EventManager.php:188 source  Laminas\EventManager\EventManager->
    triggerListeners(arguments)

    178:        public function triggerEvent(EventInterface $event)
    179:        {
    180:            return $this->triggerListeners($event);
    181:        }
    182:
    183:        /**
    184:         * @inheritDoc
    185:         */
    186:        public function triggerEventUntil(callable $callback, EventInterface $event)
    187:        {
    188:            return $this->triggerListeners($event, $callback);
    189:        }
    190:
    191:        /**
    192:         * @inheritDoc
    
      $event
    $callback

18. .../laminas/laminas-mvc/src/Application.php:331 source  Laminas\
    EventManager\EventManager->triggerEventUntil(arguments)

    321:                }
    322:            }
    323:
    324:            if ($event->getError()) {
    325:                return $this->completeRequest($event);
    326:            }
    327:
    328:            // Trigger dispatch event
    329:            $event->setName(MvcEvent::EVENT_DISPATCH);
    330:            $event->stopPropagation(false); // Clear before triggering
    331:            $result = $events->triggerEventUntil($shortCircuit, $event);
    332:
    333:            // Complete response
    334:            $response = $result->last();
    335:            if ($response instanceof ResponseInterface) {
    
    $callback
      $event

19. .../www/staging/releases/54/public/index.php:30 source  Laminas\Mvc\
    Application->run()

    20:        echo '<p>Please try to install the dependencies via: </p>';
    21:        echo '<code>cd '. realpath('.') .'<br>./install.sh</code>';
    22:        echo '<p>exit at ' . __FILE__ . ' in line ' . __LINE__ .'</p>';
    23:        exit;
    24:    }
    25:
    26:    use Core\Application;
    27:
    28:    chdir(dirname(__DIR__));
    29:    $config = __DIR__.'/../config/config.php';
    30:    Application::init(include $config)->run();
    31: 
```

**Comment by kilip** [comment]
@cbleek @TiSiE ,

I can't reproduce this issue.

I have use file transport for mail service:
```php
// /path/to/yawik/config/autoload/MailServiceOptions.config.local.php
return [
    'options' => [
        'Core/MailServiceOptions' => [
            'options' => [
                'transportClass' => 'file',
                'path' => __DIR__.'/../../.scratches/mail',
            ],
        ],
    ]
];
```

Here's a sample output
```
Date: Wed, 24 Feb 2021 10:15:30 +0100
Content-Type: text/plain;
 charset="UTF-8"
Subject: =?UTF-8?Q?Your=20application=20dated=2010/02/2020?=
To: =?UTF-8?Q?Demo=20Applicant?= <yawik-applicant@trashmail.com>
From: =?UTF-8?Q?RZ?= <yawik-demo@trash-mail.com>
Bcc: =?UTF-8?Q?Demo=20User?= <yawik-demo@trash-mail.com>
X-Mailer: php/YAWIK

Dear Mr. Applicant

Wir möchten Sie einladen
```

**Comment by kilip** [comment]
I have found that this issue is related to #627, and should be fix in #631
